flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/operations/RichTableSourceQueryOperation.java

```diff
@@ -30,17 +32,20 @@
  * A {@link TableSourceQueryOperation} with {@link FlinkStatistic} and qualifiedName.
  * TODO this class should be deleted after unique key in TableSchema is ready
  * and setting catalog statistic to TableSourceTable in DatabaseCalciteSchema is ready
+ *
+ * <p>This is only used for testing.
```

Comment:
It should not only be used for testing. It should be used for anonymous table sources that are not registered in a catalog.

Reply:
`RichTableSourceQueryOperation` is only used in `TableTestBase` for adding test TableSource with statistic. and it will not used for user interface, such as `TableEnvironmentImpl#fromTableSource`.